### PR TITLE
[PSR-6] move exists() from Item to ItemPool

### PR DESCRIPTION
**Pull request by rindeal**
related issues: #609, #389 
related discussion: https://groups.google.com/forum/#!topic/php-fig/NKaQhF98TVI
ping: @Crell  @simensen 

Wording will likely need to change as well.

### CODE REVIEW

File: proposed/cache.md [hunk 1]
```diff
@@ -315,6 +303,18 @@ interface CacheItemPoolInterface
...
+     * Confirms if the cache item exists in the cache.
+     *
+     * Note: This method MAY avoid retrieving the cached value for performance
+     * reasons, which could result in a race condition between exists() and get().
+     * To avoid that potential race condition use isHit() instead.
+     *
+     * @return boolean
+     *  True if item exists in the cache, false otherwise.
+     */
+    public function itemExists();
```
Comment (joshdifabio) [inline proposed/cache.md]: Don't we need to pass the key in here?

Comment (rindeal) [inline proposed/cache.md]: Maybe... :D

File: proposed/cache.md [hunk 2]
```diff
@@ -315,6 +303,20 @@ interface CacheItemPoolInterface
     public function getItems(array $keys = array());
 
     /**
+     * Confirms if the cache item exists in the cache.
+     *
+     * Note: This method MAY avoid retrieving the cached value for performance
+     * reasons, which could result in a race condition between exists() and get().
+     * To avoid that potential race condition use isHit() instead.
```
Comment (Crell) [inline proposed/cache.md]: The docblock also needs to be updated to clarify that we're talking about get() on Item, not on Pool.

**Comment by simensen** [comment]
@Crell @dragoonis @tedivm Are we considering this change for when we do the review/draft/review switcharoo? Do we need to discuss it more / get more feedback?

**Comment by dragoonis** [comment]
@simensen yes, no further discussion needed, I believe. Rob and myself are happy with this. I don't recall see any push back from Larry either.

Merge away!

**Comment by simensen** [comment]
@dragoonis Indeed, @Crell has actually contributed to this PR. That said, I'd still like a :+1: from @Crell but I expect that will probably be the case. Thanks for providing feedback on this!

**Comment by Crell** [comment]
The latest version above removes the race condition mention entirely, which is not what I was asking for. :smile: I said it needs to be revised to make it clear what call was being referred to.

We also should check the rest of the spec and make sure nothing else needs to be updated as well, such as the mentions of exists in the prose section.

I'm not opposed in concept, just want to make sure it's fully polished.

**Comment by Crell** [comment]
I went through the doc and didn't see anywhere else that needs updating as a result of this change.  It sounds like everyone is on board, so merging.

**Comment by dragoonis** [comment]
@rindeal ping

**Comment by Crell** [comment]
@dragoonis Can you rebase this branch to a new PR, preserving @rindeal's commit credits? I can merge it then.

**Comment by dragoonis** [comment]
@Crell I'm slammed today, totally, as I'm about to go on vacation for my birthday. Would you be able to rebase this, on this occasion? I'd appreciate it, so we can get this pushed through. I think it's the last outstanding PR.

**Comment by rindeal** [comment]
Sorry for the delay. I could do it if you tell me which commands to issue. I'm not so proficient in merging conflicts and wouldn't like to do more harm than good.

**Comment by Crell** [comment]
Merged master, filed new PR: #641.  @dragoonis, please verify and merge ASAP.

**Pull request closed without merging**